### DISCUSSION TITLE
How to apply an initial scripts in the R2dbc url

### DISCUSSION THREAD

**Discussion by hantsy** [testcontainers/testcontainers-java#4961] [Q&A]
I started a testcontaienrs postgres docker container in my tests using the following URL.

```properties
spring.r2dbc.url=r2dbc:tc:postgresql:///databasename?TC_IMAGE_TAG=12
```

How to apply an `init.sql` in this container?

**Comment by kiview** [comment]
Hey @hantsy, Testcontainers' `r2dbc`integration doesn't support the `TC_INITSCRIPT` feature of the JDBC URL integration. Since it seems you are already using Spring, I'd suggest making use of Spring's [Database Initialization support](https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto.data-initialization.using-basic-sql-scripts) or by using other general-purpose database migration libraries, such as Flyway.

**Reply by hantsy** [reply]
I have tried to use Database Initialization before, failed, [the init.sql script](https://github.com/hantsy/spring-r2dbc-sample/blob/master/testcontainers/src/test/resources/init.sql) includes functionality to create an `Enum` in multi lines.
